pkg/upgradeservice/preflight/preflight.go

```diff
@@ -142,31 +147,61 @@ func Run(app *apptypes.App, archiveDir string, sequence int64, registrySettings
 	return nil
 }
 
-func setPreflightResults(results *types.PreflightResults) error {
-	preflightData, err := getPreflightData()
+func setPreflightResults(appSlug string, results *types.PreflightResults) error {
+	preflightData, err := GetPreflightData()
 	if err != nil {
 		return errors.Wrap(err, "failed to get preflight data")
 	}
-	preflightData.Results = results
+	resultsBytes, err := json.Marshal(results)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal preflight results")
+	}
+	createdAt := time.Now()
+	preflightData.Result = &types.PreflightResult{
+		Result:                     string(resultsBytes),
+		CreatedAt:                  &createdAt,
+		AppSlug:                    appSlug,
+		ClusterSlug:                "this-cluster",
+		Skipped:                    false,
+		HasFailingStrictPreflights: hasFailingStrictPreflights(results),
+	}
+	// clear the progress once the results are set
+	preflightData.Progress = ""
```

Comment:
nit pick: can this be inside the struct above?

Reply:
made a change in https://github.com/replicatedhq/kots/pull/4693/commits/7a4d7bb34a02a9b1e766f9607cd4febd0fbfeb9e to address this one